components/rhs_thread/rhs_thread.tsx

```diff
@@ -32,40 +35,53 @@ export function renderThumbHorizontal() {
     return (<div/>);
 }
 
-export function renderThumbVertical(props) {
+export function renderThumbVertical(props: Props) {
     return (
         <div
             {...props}
             className='scrollbar--vertical'
         />);
 }
 
-export default class RhsThread extends React.Component {
-    static propTypes = {
-        posts: PropTypes.arrayOf(PropTypes.object).isRequired,
-        channel: PropTypes.object.isRequired,
-        selected: PropTypes.object.isRequired,
-        previousRhsState: PropTypes.string,
-        currentUserId: PropTypes.string.isRequired,
-        previewCollapsed: PropTypes.string.isRequired,
-        previewEnabled: PropTypes.bool.isRequired,
-        socketConnectionStatus: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            removePost: PropTypes.func.isRequired,
-            selectPostCard: PropTypes.func.isRequired,
-            getPostThread: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    posts: Post[];
+    channel: Partial<Channel> | null;
+    selected: Record<string, any>;
```

Comment:
Similarly, shouldn't this be:
```suggestion
    selected: Post;
```